app/uk/gov/hmrc/pillar2externalteststub/models/uktr/UKTRSubmissionData.scala

```diff
@@ -34,15 +34,21 @@ case class UKTRSubmissionData(
 
 object UKTRSubmissionData {
   implicit val uktrSubmissionDataFormat: OFormat[UKTRSubmissionData] = Json.format[UKTRSubmissionData]
-  val amountErrorMessage =
-    " must be Numeric, positive, with at most 2 decimal places, and less than or equal to 13 characters, including the decimal place."
-
-  private def isValidUKTRAmount(number: String): Boolean = {
-    val pattern = """^\d{1,13}\.{0,1}\d{0,2}$""".r
-    number match {
-      case pattern() if number.length <= 13 && number.toDouble > 0 => true
-      case _                                                       => false
-    }
+
+  private val boundaryUKTRAmount                    = BigDecimal("9999999999999.99")
+  private def isValidUKTRAmount(number: BigDecimal) = number >= 0 && number <= boundaryUKTRAmount && number.scale <= 2
+  private val amountErrorMessage                    = s"must be a number between 0 and $boundaryUKTRAmount with up to 2 decimal places"
+
+  private def obligationMTTRule(plrReference: String): ValidationRule[UKTRSubmissionData] = ValidationRule { data =>
+    if (data.obligationMTT && isDomesticOnly(plrReference)) {
+      invalid(
+        UKTRSubmissionError(
+          UKTRErrorCodes.REQUEST_COULD_NOT_BE_PROCESSED_003,
+          "obligationMTT",
+          "obligationMTT cannot be true for a domestic-only group or false for a non-domestic group"
```

Comment:
This error message isn't correct. ObligationMTT can be false for a non-domestic group